crates/ast/src/constrain.rs

```diff
@@ -2064,7 +2065,7 @@ pub mod test_constrain {
 
                 assert_eq!(actual_str, expected_str);
             }
-            Err(e) => panic!("syntax error {:?}", e),
+            Err(e) => internal_error!("syntax error {:?}", e),
```

Comment:
I am not very fimilare with this code, but I wonder if this is a `user_error!` that just doesn't have a pretty message yet.

Reply:
I think using `internal_error` is still okay here, things that don't exit well to the user are internal errors to the compiler.